libs/mintycore/src/main/java/com/solanamobile/mintyfresh/mintycore/usecase/PerformMintUseCase.kt

```diff
@@ -107,6 +107,12 @@ class PerformMintUseCase @Inject constructor(
 
             metadataBundle.sign(signer)
 
+            try {
+                transferTxn.setRecentBlockHash(blockhashRepository.getLatestBlockHash())
+            } catch (e: Exception) {
+                throw ExecutionException(e)
```

Comment:
Should this set the _mintState instead and return instead of throwing here. 

Doesn't seem like anything is handling this exception.

Reply:
this gets handled inside the transact block and returns a Transaction.failure from transact, which then gets caught below. changed in my most recent version tho (this is till WIP)